app/views/foreman_ansible/job_templates/run_playbook-ansible_default.erb

```diff
@@ -12,6 +12,14 @@ template_inputs:
   description: The playbook to run against given hosts
   advanced: false
   value_type: plain
+- name: check_mode
+  required: true
+  input_type: user
```

Comment:
If this is a user input, user must set/override that per run. I think we should instead make it a parameter type, so that it can be customized through parameters (e.g. on host level, host group level or set a global value). That way I can easily say all hosts should be running in the check mode until I explicitly allow the run for real perhaps on a specific host only.

We should also seed such parameter as a global parameter, the name could be `ansible_roles_check_mode` and the parameter should be of a boolean type.

Reply:
I have a few questions regarding this:
1. I managed to find only user type inputs, is there an input type parameter, should it be added or should it be implemented in another way?
2. Do you think we should also let the user set the check mode for every run or only as a host/host_group parameter? and if so what will be prioritized first when running the host roles for example?

Reply:
host parameter doesn't need to be specified in the template metadata, you get it by `host_param` from the template.
https://github.com/theforeman/foreman/blob/ccbc96d8f2aa962a72b9b1ede33a657504723c2c/app/models/concerns/host_params.rb#L52

If we seed it as global parameter, we may even rely on it being present.

Being able to override this on a template level may be useful in some situations as well.
In my opininion if we choose to have such template input, it should not be required, but if filled, it should take precedence.

Reply:
@ezr-ondrej So just to make sure I understood, we want the templates to have a user input that is not required and also we want to have a global parameter to a host or host group. If the user filled the option in the job it should get precedence and if not it will get the check mode value from the global parameter. Is this correct?

Reply:
Yes this is correct :+1:
The template option is not even mandatory, so if you'd struggle with the parameter/option precedence, we can end up just with the parameter and it would be ok. But having both is the best result :+1: 

*Just note (to be sure we are talking about the same parameters):* it is always host parameter, it just can get inherited from Hostgroup, but also OS, org/loc, or just from global values.

Reply:
Ok thanks for the clarification, I will fix it.